webxrlayers-1.bs

```diff
@@ -929,17 +989,14 @@ To <dfn>allocate color textures</dfn> using an {{XRCompositionLayer}} |layer|, a
         <dt> Otherwise
           <dd> Initialize |array| with 1 [=new=] instance of {{WebGLTexture}} in the [=relevant realm=] of |context| created as a {{TEXTURE_2D}} texture with |context| and |init|'s {{XRLayerInit/alpha}}, {{XRLayerInit/viewPixelWidth}} and {{XRLayerInit/viewPixelHeight}} values.
         <dl>
-  1. let |numViews| be the number of all the session's {{XRView|XRViews}}.
+  1. let |numViews| be the number of the |session|'s [=list of views=].
   1. If |layer|'s {{XRCompositionLayer/layout}} is {{XRLayerLayout/"default"}}:
         <dl class="switch">
         <dt> If |textureType| is {{"texture-array"}}:
-          <dd> Initialize |array| with 1 [=new=] instance of {{WebGLTexture}} in the [=relevant realm=] of |context| created as a {{TEXTURE_2D_ARRAY}} texture with |numViews| internal textures using |context| and |init|'s {{XRLayerInit/alpha}}, {{XRLayerInit/viewPixelWidth}} and {{XRLayerInit/viewPixelHeight}} values.
+          <dd> Initialize |array| with 1 [=new=] instance of {{WebGLTexture}} in the [=relevant realm=] of |context| created as a {{TEXTURE_2D_ARRAY}} texture with |numViews| layers using |context| and |init|'s {{XRLayerInit/alpha}}, {{XRLayerInit/viewPixelWidth}} and {{XRLayerInit/viewPixelHeight}} values.
           <dd> Return |array| and abort these steps.
         <dt> Otherwise
-          <dd> For each |view| in the |session|'s [=list of views=]:
-            1. Let |glViewport| be the [=WebGL viewport=] from the [=list of viewports=] associated with |view|.
-            1. let |texture| be a [=new=] instance of {{WebGLTexture}} in the [=relevant realm=] of |context| created as a {{TEXTURE_2D}} texture with |context|, |init|'s {{XRLayerInit/alpha}} and |glViewport|'s width and height.
-            1. Append |texture| to |array|.
+          <dd> Initialize |array| with |numViews| [=new=] instances of {{WebGLTexture}} in the [=relevant realm=] of |context| created as a {{TEXTURE_2D}} texture with |context| and |init|'s {{XRProjectionLayerInit/alpha}}, {{XRLayerInit/viewPixelWidth}} and {{XRLayerInit/viewPixelHeight}} values.
```

Comment:
This doesn't support different texture sizes for different views. Should it?

Reply:
This is the algorithm for non-projection layers.
The size for the layer is given during the initialization and is the same for each view.

Reply:
Hmm, so for a device with a high-resolution HMD and a low-resolution camera, we're going to use the same texture size for all of them?

Reply:
yes